packages/frontend/core/src/modules/quicksearch/services/cmdk.ts

```diff
@@ -0,0 +1,82 @@
+import type { DocsService } from '@toeverything/infra';
+import { Service } from '@toeverything/infra';
+
+import type { WorkbenchService } from '../../workbench';
+import { CollectionsQuickSearchSession } from '../impls/collections';
+import { CommandsQuickSearchSession } from '../impls/commands';
+import { CreationQuickSearchSession } from '../impls/creation';
+import { DocsQuickSearchSession } from '../impls/docs';
+import { RecentDocsQuickSearchSession } from '../impls/recent-docs';
+import type { QuickSearchService } from './quick-search';
+
+export class CMDKQuickSearchService extends Service {
+  constructor(
+    private readonly quickSearchService: QuickSearchService,
+    private readonly workbenchService: WorkbenchService,
+    private readonly docsService: DocsService
+  ) {
+    super();
+  }
+
+  toggle() {
+    if (this.quickSearchService.quickSearch.show$.value) {
+      this.quickSearchService.quickSearch.hide();
+    } else {
+      this.quickSearchService.quickSearch.show(
+        [
+          this.framework.createEntity(RecentDocsQuickSearchSession),
```

Comment:
I think these sessions should be put as groups of config in quickSearch service, and using a strategy pattern with key identifiers to select the appropriate session group for `show`.

Reply:
The reason session provided by the caller is that the source of these sessions is very flexible.

For example, in insert link search, any third-party link can be matched. For this kind of non-reusable and scenario-specific, I can use a temporary function to implement it.

https://github.com/toeverything/AFFiNE/pull/7214/files/2c0c6bc958d7885dc4bd6d6854911da3cd10ab38#diff-54ad81614a441711f595238131eb727b2c0e7bd0aa8ecfb4c83c121ba2305493R376

Implementing it in QuickSearchService may make it unclear due to lack of context